src/app/app-interactive-map/store/interactive-map.effects.ts

```diff
@@ -131,13 +151,9 @@ export class InteractiveMapEffects {
     map(([action, storeState]: [fromActions.SetModel, AppState]) => {
       const model = action.payload;
       const {maps} = storeState.interactiveMap;
-      const modelHasMaps = (mapItem: types.MapItem) =>
-        (mapItem.model === model.name);
-      const preferredMap = (mapItem: types.MapItem) =>
-        (mapItem.model === preferredMapItem.model && mapItem.name === preferredMapItem.name);
-      const predicate = (mapItem: types.MapItem) =>
-        (mapItem.model === model.name && mapItem.name === preferredMapsByModel[model.name]);
-      return new fromActions.SetMap(preferredSelector(predicate, preferredMap, modelHasMaps)(maps));
+      const preferred = maps.filter((item) => compareModel(item, preferredMapItem) && compareName(item, preferredMapItem));
+      const selectedMap = selectorCreator([compareModel, compareName])(maps, preferredMapsByModel[model.name] || {model: model.name}, preferred);
```

Comment:
{model: model.name} for filter by model if the preferred doesn't exists for that model.